src/dune_cache/local.ml

```diff
@@ -37,7 +37,7 @@ module Target = struct
     | { Unix.st_kind = Unix.S_REG; st_perm; _ } ->
       Path.Build.chmod path
         ~mode:(Path.Permissions.remove Path.Permissions.write st_perm);
-      let executable = st_perm land 0o100 <> 0 in
+      let executable = Path.Permissions.test Path.Permissions.execute st_perm in
       Some { path; executable }
     | (exception _)
```

Comment:
Not related to this PR, but shouldn't we just catch Unix errors here?

Reply:
I'm not sure. I think the idea for ignoring all errors here is that we don't want to break builds because the shared cache has some issues. Do you think it's better to fail hard here?

Reply:
I guess one could argue that instead of silently ignoring errors here, it would be better to log them.